README.md

```diff
@@ -257,6 +257,49 @@ codacy-analysis-cli analyse \
 
 > In alternative to setting `--api-token` you can define CODACY_API_TOKEN in the environment.
 
+## GitHub Action
+
+The analysis CLI is available to be executed as a GitHub Action. The following is an example of a workflow
+using the CLI as an action, to analyse each commit and pull request, with your Codacy configuration.
+
+```yaml
+name: codacy-analysis-cli
+
+on: ["push"]
+
+jobs:
+  codacy-analysis-cli:
+    runs-on: ubuntu-latest
+    name: codacy-analysis-cli
+    steps:
+      - uses: actions/checkout@master
+      - name: Run codacy-analysis-cli
+        uses: codacy/codacy-analysis-cli@master
+        with:
+          project-token: ${{ secrets.CODACY_PROJECT_TOKEN }}
+```
+
+Running the action with the default configurations will:
+
+- Analyse the current commit or pull request, using your Codacy configurations, running all supported
+    tools for the languages you are using.
+- Print analysis results into the console (you can check them in GitHub's action workflow panel).
+- Fail the workflow if at least one issue is found in your code.
+
+Check the next section to see how you can further configure this action.
+
+### Caveats
+
+This action supports all [CLI configuration options](#commands-and-configuration) with the following exceptions:
+
+- `--codacy-api-base-url` -- **Not supported**. The action will only get your configurations from Codacy cloud.
+- `--commit-uuid` -- **Not supported**. The action will only analyse the commit that triggered it.
+- `--api-token` -- **Not supported**. Use [`--project-token`](#project-token) instead.
+- `--username` -- **Not supported**. Use [`--project-token`](#project-token) instead.
+- `--project` -- **Not supported**. Use [`--project-token`](#project-token) instead.
+
+The command `validate-configuration` is also **not supported** since only remote configuration is allowed.
```

Comment:
Should we allow for local configuration files? This might not be as straightforward as it seems, due to the lack of conditionals in the action yaml.

For example, supporting this means that `project-token` is optional. But the way we deal with that, in the action yaml, is to have a default value instead, which in this case does not make sense.

Reply:
Well the conditionals are now solved, so we should decide if allowing local configuration files is ok.

Reply:
I think i now understood what you mean by local configs.
We will always use local configs if they exists either way, there is no way to disable it.

This `validate-configuration` command is just to validate the config before doing an analysis.
Either way, possibly this command is not very important in this context since you would probably run it locally.